### ISSUE TITLE
Restructure k8schain imports

### ISSUE THREAD

**Issue by jonjohnsonjr** [google/go-containerregistry#1042]
We do side-effectual imports in k8schain to pull in cloud-specific credential providers. We currently have build tags to disable importing specific things in k8schain. This seems backwards.

Callers should be responsible for importing these magic packages, not k8schain, which would make the dependency graph a little less stupid.

**Comment by jonjohnsonjr** [comment]
@dprotaso @hasheddan WDYT

**Comment by dprotaso** [comment]
Yeah I agree - this could be coupled with adoption of the new k8s cloud provider libs
